src/ExcelServiceProvider.php

```diff
@@ -86,6 +86,10 @@ public function register()
             return new Filesystem($this->app->make('filesystem'));
         });
 
+        $this->app->singleton(Writer::class, function ($app) {
```

Comment:
Can you explain why this change is needed, I'm a bit cautious with this change as it might have side effects.

Reply:
Yes, I made this change because since the Writer class was not originally bound as a singleton(or instance), we would get a new instance each time it was resolved , meaning that all the listeners were gone from the object, which is why an event raised in the write method(which is called multiple times) would fire only once.

However, you are right to be cautious since this seemingly little change introduces a major bug when we have multiple queued exports running at the same time.

I will have to come up with another solution, but if you have an idea it would love to hear it.